packages/reactive-element/src/decorators/custom-element.ts

```diff
@@ -13,6 +13,125 @@
 
 import type {Constructor} from './base.js';
 
+declare global {
+  interface CustomElementRegistry {
+    getName: (ctor: CustomElementConstructor) => string | null;
+  }
+}
+
+export type DefineErrorCallback = (
+  error: Error,
+  tagName: string,
+  ctor: CustomElementConstructor
+) => void;
+
+/**
+ *
+ * @param tagName The tag name of the custom element to define.
+ * @param ctor The constructor of the custom element to define
+ * @param onError An optional function which can handle a define exception
+ * error. If provided, the exception is caught.
+ * @param onError
+ * @returns void
+ */
+export const define = (
```

Comment:
This could be used by users that aren't using the decorator but they'd have to import it from `lit/decorators/custom-element.js` or `lit/decorators.js`.

Would it maybe make sense to move `define`, `warn`, and `warnPackabeVErsion` to a separate submodule like `@lit/reactive-element/define.js`, which will be mirrored in `lit-element` and `lit`?

Reply:
I think we should do exactly that.